java/src/main/java/com/genexus/util/GXStorageProvider.java

```diff
@@ -107,7 +108,7 @@ public boolean get(String storageobjectfullname, GXFile externalFile, GXBaseColl
     public boolean getPrivate(String storageobjectfullname, GXFile externalFile, int expirationMinutes, GXBaseCollection<SdtMessages_Message> messages) {
         try {
             validProvider();
-            String url = provider.get(storageobjectfullname, true, expirationMinutes);
+            String url = provider.get(storageobjectfullname, ResourceAccessControlList.Private, expirationMinutes);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 109:121, 92:104.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.